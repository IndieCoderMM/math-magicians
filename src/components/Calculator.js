import React from 'react';

// eslint-disable-next-line
class Calculator extends React.Component {
  render() {
    return (
      <div className="calculator-grid">
        <div className="output-screen">
          <p className="output-text">0</p>
        </div>
        <button type="button" className="operator">
          AC
        </button>
        <button type="button" className="operator">
          +/-
        </button>
        <button type="button" className="operator">
          %
        </button>
        <button type="button" className="operator">
          /
        </button>
        <button type="button">7</button>
        <button type="button">8</button>
        <button type="button">9</button>
        <button type="button" className="operator">
          x
        </button>
        <button type="button">4</button>
        <button type="button">5</button>
        <button type="button">6</button>
        <button type="button" className="operator">
          -
        </button>
        <button type="button">1</button>
        <button type="button">2</button>
        <button type="button">3</button>
        <button type="button" className="operator">
          +
        </button>
        <button type="button">0</button>
        <button type="button">.</button>
        <button type="button" className="operator wide">
          =
        </button>
      </div>
    );
  }
}

export default Calculator;
