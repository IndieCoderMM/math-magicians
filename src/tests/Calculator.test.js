import TestRenderer from 'react-test-renderer';
import Calculator from '../components/Calculator';

describe('Calculator Component', () => {
  it('should render correctly', () => {
    const tree = TestRenderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
