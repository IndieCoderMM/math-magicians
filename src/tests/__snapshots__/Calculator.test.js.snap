// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator Component should render correctly 1`] = `
<div
  className="calculator-grid"
>
  <div
    className="output-screen"
  >
    <div
      className="formula"
    />
    <div
      className="output-text"
    >
      0
    </div>
  </div>
  <button
    onClick={[Function]}
    type="button"
  >
    AC
  </button>
  <button
    onClick={[Function]}
    type="button"
  >
    +/-
  </button>
  <button
    onClick={[Function]}
    type="button"
  >
    %
  </button>
  <button
    className="operator"
    onClick={[Function]}
    type="button"
  >
    รท
  </button>
  <button
    onClick={[Function]}
    type="button"
  >
    7
  </button>
  <button
    onClick={[Function]}
    type="button"
  >
    8
  </button>
  <button
    onClick={[Function]}
    type="button"
  >
    9
  </button>
  <button
    className="operator"
    onClick={[Function]}
    type="button"
  >
    x
  </button>
  <button
    onClick={[Function]}
    type="button"
  >
    4
  </button>
  <button
    onClick={[Function]}
    type="button"
  >
    5
  </button>
  <button
    onClick={[Function]}
    type="button"
  >
    6
  </button>
  <button
    className="operator"
    onClick={[Function]}
    type="button"
  >
    -
  </button>
  <button
    onClick={[Function]}
    type="button"
  >
    1
  </button>
  <button
    onClick={[Function]}
    type="button"
  >
    2
  </button>
  <button
    onClick={[Function]}
    type="button"
  >
    3
  </button>
  <button
    className="operator"
    onClick={[Function]}
    type="button"
  >
    +
  </button>
  <button
    className="wide-btn"
    onClick={[Function]}
    type="button"
  >
    0
  </button>
  <button
    onClick={[Function]}
    type="button"
  >
    .
  </button>
  <button
    className="operator"
    onClick={[Function]}
    type="button"
  >
    =
  </button>
</div>
`;
